File: adk/tests/fixtures/integration_profiles/node-integration/project/src/repository.ts
export class InventoryRepository {
  private readonly stock: Map<string, number>;

  constructor(stock: Map<string, number>) {
    this.stock = stock;
  }

  reserve(sku: string, quantity: number): boolean {
    const available = this.stock.get(sku) ?? 0;
    if (available < quantity) return false;
    this.stock.set(sku, available - quantity);
    return true;
  }
}

File: adk/tests/fixtures/integration_profiles/python-integration/project/app/repository.py
class InventoryRepository:
    def __init__(self, stock: dict[str, int]):
        self._stock = stock

    def reserve(self, sku: str, quantity: int) -> bool:
        available = self._stock.get(sku, 0)
        if available < quantity:
            return False
        self._stock[sku] = available - quantity
        return True
